src/lib/state.ts

```diff
@@ -83,6 +83,24 @@ export type PersistenceState = ({
 	stale: boolean
 }
 
+export enum RoomStatus {
+	CONNECTED,
+	CONNECTING,
+	DISCONNECTED
+}
+
+export type RoomParticipant = {
+	userEmail: string
+	isHost: boolean
+}
+
+export type RoomState = {
+	status: RoomStatus
+	roomId: string
+	password: string
```

Comment:
do we want this in plaintext?  we could keep this encrypted + salt

Reply:
I don't think it's super important to hash it, and it's not yet used.